zcash_client_backend/src/data_api/wallet/input_selection.rs

```diff
@@ -239,6 +239,75 @@ impl<FeeRuleT, NoteRef> Proposal<FeeRuleT, NoteRef> {
     pub fn is_shielding(&self) -> bool {
         self.is_shielding
     }
+
+    /// Returns whether or not this proposal requires interaction with the specified pool
+    //
+    // TODO: the `TransactionRequest` type doesn't contain quite enough information about
+    // recipients, because it does not resolve each output payment address to a single receiver.
+    // In the presence of source restriction metadata or pool-crossing prohibitions, we need to
+    // resolve receiving pools prior to constructing the proposal; otherwise we don't have
+    // sufficient information to fully determine the privacy implications of a transaction.
+    pub fn involves(&self, pool_type: PoolType) -> bool {
+        match pool_type {
+            PoolType::Transparent => {
+                self.is_shielding
+                    || !self.transparent_inputs.is_empty()
+                    || self.transaction_request.payments().iter().any(|p| {
+                        match &p.recipient_address {
+                            Address::Sapling(_) => false,
+                            Address::Transparent(_) => true,
+                            Address::Unified(addr) => {
+                                !addr.has_orchard() && !addr.has_sapling() && addr.has_transparent()
```

Comment:
Please add a more specific comment about this line (if it's not fixed before this merges).

Reply:
What needs to happen here is that after #1187, this should check that the transparent component of the UA is used.